miden-tx/src/host/account_procs.rs

```diff
@@ -10,67 +13,74 @@ use crate::errors::TransactionHostError;
     /// Returns a new [AccountProcedureIndexMap] instantiated with account procedures present in
     /// the provided advice provider.
     pub fn new(
-        account_code_commitment: Digest,
+        native_account_code_commitment: Digest,
         adv_provider: &impl AdviceProvider,
+        foreign_account_code_commitments: &[Digest],
     ) -> Result<Self, TransactionHostError> {
-        // get the account procedures from the advice_map
-        let proc_data =
-            adv_provider.get_mapped_values(&account_code_commitment).ok_or_else(|| {
-                TransactionHostError::AccountProcedureIndexMapError(
-                    "Failed to read account procedure data from the advice provider".to_string(),
-                )
-            })?;
-
         let mut result = BTreeMap::new();
 
-        // sanity checks
-
-        // check that there are procedures in the account code
-        if proc_data.is_empty() {
-            return Err(TransactionHostError::AccountProcedureIndexMapError(
-                "The account code does not contain any procedures.".to_string(),
-            ));
-        }
-
-        // check that procedure data have a correct length
-        if proc_data.len() % AccountProcedureInfo::NUM_ELEMENTS_PER_PROC != 0 {
-            return Err(TransactionHostError::AccountProcedureIndexMapError(
-                "The account procedure data has invalid length.".to_string(),
-            ));
-        }
-
-        // One procedure requires 8 values to represent
-        let num_procs = proc_data.len() / AccountProcedureInfo::NUM_ELEMENTS_PER_PROC;
-
-        // check that the account code does not contain too many procedures
-        if num_procs > AccountCode::MAX_NUM_PROCEDURES {
-            return Err(TransactionHostError::AccountProcedureIndexMapError(
-                "The account code contains too many procedures.".to_string(),
-            ));
-        }
-
-        for (proc_idx, proc_info) in
-            proc_data.chunks_exact(AccountProcedureInfo::NUM_ELEMENTS_PER_PROC).enumerate()
+        for code_commitment in
+            foreign_account_code_commitments.iter().chain(&[native_account_code_commitment])
         {
-            let proc_info_array: [Felt; AccountProcedureInfo::NUM_ELEMENTS_PER_PROC] =
-                proc_info.try_into().expect("Failed conversion into procedure info array.");
-
-            let procedure = AccountProcedureInfo::try_from(proc_info_array).map_err(|e| {
-                TransactionHostError::AccountProcedureIndexMapError(format!(
-                    "Failed to create AccountProcedureInfo: {:?}",
-                    e
-                ))
+            // get the account procedures from the advice_map
+            let proc_data = adv_provider.get_mapped_values(code_commitment).ok_or_else(|| {
+                TransactionHostError::AccountProcedureIndexMapError(
+                    "Failed to read account procedure data from the advice provider".to_string(),
+                )
```

Comment:
Would it make sense to have a single helper functions that given a code commitment returns a map? For example, something like:
```rust
fn build_account_procedure_map(
    code_commitment: Digest, 
    adv_provider: &impl AdviceProvider,
) -> Result<BTreeMap<Digest, u8>, TransactionHostError>
```